### ISSUE TITLE
[bug] NUnit2010 should provide a different suggestion for null comparison

### ISSUE THREAD

**Issue by Bartleby2718** [nunit/nunit.analyzers#746]
Suppose the following:
```cs
string? nullString = null;
string? nonNullString = "nonNullString";
int? nullInt = null;
int? nonNullInt = 42;
```
Then, NUnit2010 flags the following assertions and proposes some fixes (as of `NUnit.Analyzers` 4.2.0):
```cs
// Assert.That(nullString, Is.EqualTo(null)); is proposed for the following two:
Assert.That(nullString is null);
Assert.That(nullString == null);

// Assert.That(nonNullString, Is.Not.EqualTo(null)); is proposed for the following two:
Assert.That(nonNullString is not null);
Assert.That(nonNullString != null);

// Assert.That(nullInt, Is.EqualTo(null)); is proposed for the following two:
Assert.That(nullInt is null);
Assert.That(nullInt == null);

// Assert.That(nonNullInt, Is.Not.EqualTo(null)); is proposed for the following two:
Assert.That(nonNullInt is not null);
Assert.That(nonNullInt != null);
```

I don't know how [`Is.EqualTo`](https://docs.nunit.org/articles/nunit/writing-tests/constraints/EqualConstraint.html) is implemented under the hood, but I'm pretty sure `Is.Null` or `Is.Not.Null` should be used for either half or all of the fixes, depending on how `Is.EqualTo` works.

**Comment by manfred-brands** [comment]
The actual generated code is correct, but indeed it could improve the code more by checking the actual argument tested and generated `Is.Null` for `null` and also `Is.Zero` for `0`.

**Comment by mikkelbu** [comment]
Currently, NUnit2010 changes `Assert.That(nonNullInt is not null);` into `Assert.That(nonNullInt, Is.Not.EqualTo(null));` and then we can use NUnit4002 to change it into `Assert.That(nonNullInt, Is.Not.Null);` (and similarly for the other cases, but I agree that NUnit2010 could just as well combine both refactorings into one.
